spiffe-helper_test.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"crypto/x509"
+	"github.com/spiffe/spire/proto/api/workload"
+	"github.com/spiffe/spire/test/util"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+	"time"
+)
+
+//Creates a Sidecar with a Mocked WorkloadAPIClient and tests that
+//running the Sidecar Daemon, when a SVID Response is sent to the
+//UpdateChan on the WorkloadAPI client, the PEM files are stored on disk
+func TestSidecar_RunDaemon(t *testing.T) {
+
+	tmpdir, err := ioutil.TempDir("", "test-certs")
+	if err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	config := &SidecarConfig{
+		Cmd:                "echo",
+		CertDir:            tmpdir,
+		SvidFileName:       "svid.pem",
+		SvidKeyFileName:    "svid_key.pem",
+		SvidBundleFileName: "svid_bundle.pem",
+	}
+
+	updateMockChan := make(chan *workload.X509SVIDResponse, 1)
+	workloadClient := MockWorkloadClient{
+		mockChan: updateMockChan,
+	}
+
+	sidecar := sidecar{
+		config:            config,
+		workloadAPIClient: workloadClient,
+	}
+
+	go func() {
+		err = sidecar.RunDaemon()
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	x509SvidTestResponse := x509SvidResponse(t)
+
+	//send a X509SVIDResponse to Updates channel
+	updateMockChan <- x509SvidTestResponse
+
+	//sleep this routine so the dumpBundles on the Sidecar is run
+	time.Sleep(1 * time.Millisecond)
```

Comment:
tests that rely on sleeps tend to be flakey. is there another way we could wait until the update has been processed?

One way would be to:
1) Pass a cancel-able context into RunDaemon
2) Modify updateMockChan to be unbuffered
3) After you send the response into updateMockChan, cancel the context and wait for RunDaemon to return

Reply:
Thank for all your feedback! I think I addressed all the comments